api/genericresource/parse.go

```diff
@@ -0,0 +1,35 @@
+package genericresource
+
+import (
+	"strconv"
+	"strings"
+
+	"github.com/docker/swarmkit/api"
+)
+
+// Parse parses the GenericResource resources given by the arguments
+func Parse(tprs string, resources *api.Resources) error {
+	resources.Generic = make(map[string]*api.GenericResource)
+	mtpr := resources.Generic
+
+	for _, term := range strings.Split(tprs, ";") {
```

Comment:
Are you saying that the function needs to be split in multiple ones ?
Or are you saying that I need to add more flags ?